### PR TITLE
Polls: use new event replacement types

### PR DESCRIPTION
**Pull request by julioromano**
From the changes in: https://github.com/ruma/ruma/pull/1638

Will allow polls edited from legacy Elements to be read rust-sdk based clients.

### CODE REVIEW

File: bindings/matrix-sdk-ffi/src/room.rs [hunk 1]
```diff
@@ -445,8 +446,10 @@ impl Room {
             .fold(question, |acc, (index, answer)| format!("{acc}\n{}. {answer}", index + 1));
 
         let poll_start_event_content =
-            UnstablePollStartEventContent::plain_text(fallback_text, poll_content_block);
-        let event_content = AnyMessageLikeEventContent::UnstablePollStart(poll_start_event_content);
+            NewUnstablePollStartEventContent::plain_text(fallback_text, poll_content_block);
+        let event_content = AnyMessageLikeEventContent::UnstablePollStart(
+            UnstablePollStartEventContent::New(poll_start_event_content),
```
Comment (zecakeh) [inline bindings/matrix-sdk-ffi/src/room.rs:451]:
```suggestion
           poll_start_event_content.into(),
```

This should word

File: crates/matrix-sdk-ui/src/timeline/tests/polls.rs [hunk 2]
```diff
@@ -201,7 +202,7 @@ impl TestTimeline {
 
     async fn send_poll_start(&self, sender: &UserId, content: UnstablePollStartContentBlock) {
         let event_content = AnyMessageLikeEventContent::UnstablePollStart(
-            UnstablePollStartEventContent::new(content),
+            UnstablePollStartEventContent::New(NewUnstablePollStartEventContent::new(content)),
```
Comment (zecakeh) [inline crates/matrix-sdk-ui/src/timeline/tests/polls.rs:205]:
```suggestion
            NewUnstablePollStartEventContent::new(content).into(),
```

Same here

File: crates/matrix-sdk-ui/src/timeline/tests/polls.rs [hunk 3]
```diff
@@ -213,7 +214,7 @@ impl TestTimeline {
         content: UnstablePollStartContentBlock,
     ) {
         let event_content = AnyMessageLikeEventContent::UnstablePollStart(
-            UnstablePollStartEventContent::new(content),
+            UnstablePollStartEventContent::New(NewUnstablePollStartEventContent::new(content)),
```
Comment (zecakeh) [inline crates/matrix-sdk-ui/src/timeline/tests/polls.rs:217]:
```suggestion
            NewUnstablePollStartEventContent::new(content).into(),
```
Same here

File: crates/matrix-sdk-ui/src/timeline/tests/polls.rs [hunk 4]
```diff
@@ -243,12 +244,11 @@ impl TestTimeline {
...
-        let mut content = UnstablePollStartEventContent::new(content);
-        content.relates_to = Some(Relation::Replacement(Replacement::new(
-            original_id.to_owned(),
-            content.clone().into(),
-        )));
-        let event_content = AnyMessageLikeEventContent::UnstablePollStart(content);
+        let content =
+            ReplacementUnstablePollStartEventContent::new(content, original_id.to_owned());
+        let event_content = AnyMessageLikeEventContent::UnstablePollStart(
+            UnstablePollStartEventContent::Replacement(content),
```
Comment (zecakeh) [inline crates/matrix-sdk-ui/src/timeline/tests/polls.rs:250]:
```suggestion
           content.into(),
```
Same here

**Comment by julioromano** [comment]
@zecakeh works indeed, but on my side how can I easily notice when a type supports from/into do I have always to read all of its source code in search of `impl From`s ?

**Comment by zecakeh** [comment]
> @zecakeh works indeed thanks for the hint! But from my side, how can I easily notice when a type supports from/into? Do I have always to read all of its source code in search of `impl From`s ?

It is in the docs: https://docs.ruma.io/ruma/events/poll/unstable_start/struct.ReplacementUnstablePollStartEventContent.html#impl-From%3CReplacementUnstablePollStartEventContent%3E-for-UnstablePollStartEventContent

**Comment by julioromano** [comment]
Keeping this as a draft until the relative ruma commit is available in the `lax-id-validation` branch.

**Comment by zecakeh** [comment]
I just pushed the commits to `lax-id-validation`, they are available at `4e8afe00229d961d206b54a61ab29f4ba1f5b2c3`

**Review by zecakeh — APPROVED** [review]

**Review by jplatte — APPROVED** [review]

**Comment by julioromano** [comment]
Android SDK build kicked off at: https://github.com/matrix-org/matrix-rust-components-kotlin/actions/runs/6220437558

**Pull request merged**